src/Microsoft.DotNet.CoreFxTesting/build/core/coverage/Coverage.targets

```
@@ -7,22 +7,26 @@
     <CoverageExecutablePath Condition="'$(CoverageExecutablePath)' == ''">$(GlobalToolsDir)coverlet</CoverageExecutablePath>
     <RunArguments>"$(TestAssembly)" --target "$(RunCommand)" --targetargs "$(RunArguments)" --format "$(CoverageFormat)" --output "$(CoverageOutputPath)" --threshold "$(CoverageThreshold)" --threshold-type "$(CoverageThresholdType)"</RunArguments>
     <RunArguments Condition="'$(CoverageSourceLink)' == 'true'">$(RunArguments) --use-source-link</RunArguments>
-    <RunArguments Condition="'$(CoverageExclude)' != ''">$(RunArguments) --exclude "$(CoverageExclude)"</RunArguments>
     <RunCommand>$(CoverageExecutablePath)</RunCommand>
   </PropertyGroup>
 
   <Target Name="SetupCoverageFilter">
 
-    <PropertyGroup>
-      <CoverageExcludeByFile>--exclude-by-file @(CoverageExcludeFile -> '"%(Identity)"', ' --exclude-by-file ')</CoverageExcludeByFile>
-      <RunArguments>$(RunArguments) $(CoverageExcludeByFile)</RunArguments>
+    <PropertyGroup Condition="'@(CoverageExcludeFile -> Count())' &gt; 0">
+      <CoverageExcludeByFileFilter>--exclude-by-file @(CoverageExcludeFile -> '"%(Identity)"', ' --exclude-by-file ')</CoverageExcludeByFileFilter>
```

Wouldn't this add a --exclude-by-file flag at the end of the RunArguments without a filter? It seems that what you want to achieve should be inverted:
```suggestion
      <CoverageExcludeByFileFilter>@(CoverageExcludeFile -> ' --exclude-by-file  ', '"%(Identity)"')</CoverageExcludeByFileFilter>
```

---

no the syntax is correct.